cdap-ui/app/cdap/components/PipelineList/DeployedPipelineView/index.tsx

```
@@ -29,10 +29,22 @@ import LoadingSVGCentered from 'components/LoadingSVGCentered';
 import { getCurrentNamespace } from 'services/NamespaceStore';
 import { gql } from 'apollo-boost';
 import { useQuery } from '@apollo/react-hooks';
+import Page500 from 'components/500';
+import ErrorBanner from 'components/ErrorBanner';
+import If from 'components/If';
 
 import './DeployedPipelineView.scss';
 import { objectQuery } from 'services/helpers';
 
+// For errors of this type coming from graphQl, we show a 500 error instead of a banner.
+const HIGH_PRIORITY_ERRORS = new Set(['pipelinesList']);
+
+interface IErrorForDisplay {
```

Do/will we use GraphQL in other locations? Should processing of errors be centralized somewhere?

---

I remember we were floating the idea of using it in other places but, currently, we do not use it anywhere else.
I assume you mean just the parsing/categorizing of errors and not showing the errors referring to processing of the errors.
We can extract the parsing to a common location but identifying the high priority ones and deciding which ones to show would be specific to each location, right? I can move that logic to a central location, let me know what you think or if I'm missing something.

---

It would make sense to move this logic to either ErrorBoundary or we can create a new Boundary component for graphql. When we migrate in other places we can throw an error from the render functions with right error objects which these boundary components can catch and show appropriate message.